File: 8kyu/filter-out-the-geese/filter-out-the-geese.js
/* 
PREP:
- use includes() to check if the birds array has geese or not
- filter out any birds that are not part of geese
*/

function gooseFilter(birds) {
	const geese = [
		"African",
		"Roman Tufted",
		"Toulouse",
		"Pilgrim",
		"Steinbacher",
	];
	return birds.filter((elem) => !geese.includes(elem));
}

File: 8kyu/filter-out-the-geese/filter-out-the-geese.py
from typing import List

geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]
def goose_filter(birds: List[str]) -> List[str]:
  return list(filter(lambda bird: bird not in geese, birds))

# def goose_filter(birds: List[str]) -> List[str]:
#   return [bird for bird in birds if bird not in geese]


print(goose_filter(["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]))
print(goose_filter(["Mallard", "Barbary", "Hook Bill", "Blue Swedish", "Crested"]))